src/app/components/screen-saver/screen-saver-header/screen-saver-header.component.ts

```diff
@@ -7,9 +9,18 @@ import { Component, OnInit } from '@angular/core';
 })
 export class ScreenSaverHeaderComponent implements OnInit {
 
-  constructor() { }
+  content: any;
+  constructor(public macroscopeDataService: MacroscopeDataService) {
+    this.content = {};
+   }
 
   ngOnInit() {
+    this.macroscopeDataService.fetchAndParseCsv('assets/macroscope-ui-descriptions.csv').subscribe((data) => {
```

Comment:
Should be updated to use service.data. Save subscription!!!! This is a memory leak!